src/CMakeLists.txt

```diff
@@ -265,13 +271,19 @@ add_library(
   $<TARGET_OBJECTS:dlaf.solver_object>
 )
 target_link_libraries(DLAF PUBLIC dlaf.prop)
+target_link_libraries(DLAF PRIVATE dlaf.prop_private)
 target_add_warnings(DLAF)
 
 # ----- DEPLOY
 include(GNUInstallDirs)
 
+# TODO: dlaf.prop_private has to be exported even if linked privately to DLAF.
```

Comment:
This is an open question which I'm happy to leave as it is. The alternative option is to directly add the flag to the targets (with some helper function or similar).

Reply:
I don't have a strong preference here, the current solutions is good enough for me.

Reply:
Actually, CMake still does not provide a good method for dealing with this.


```cmake
# Create imported target DLAF::dlaf.prop_private
add_library(DLAF::dlaf.prop_private INTERFACE IMPORTED)

set_target_properties(DLAF::dlaf.prop_private PROPERTIES
  INTERFACE_COMPILE_OPTIONS "-gz"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_OPTIONS "-gz"
)
```

Even if it is not exactly clean, I agree with you that we can leave it as you did it.

If someone decides to use `dlaf.prop_private` target in the final application, it is pretty straightforward from the name of the target to judge how risky it could be (currently it is not much anyway).

Reply:
Thanks for the comments. I've removed the TODO for now. Waiting for ok from @rasolca.

Reply:
Ok